docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,684 @@
+
+We have seen together how to test your extension and sometimes things don't go so well and you want to debug your tests.
+Same principle here, the trick is to enable the Maven Surfire remote debugging in order to attach an IDE `Remote` configuration.
+
+[source, shell]
+-------------
+$ cd `./greeting`
+$ mvn clean test -Dmaven.surefire.debug
+-------------
+
+By default, Maven will wait for a connection on `localhost:5005`.
+
+=== Extension publication
+
+Now that you just finish to build your first extension you should be eager to use it in a Quarkus application! 
+
+*Classic Maven publication*
+
+Because your extension produces a classic JAR, the easiest way to share your extension is to publish it to a Maven repository.
+Once published you can simply declare it with your project dependencies
+```
+...
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>${quarkus.platform.group-id}</groupId>
+                <artifactId>${quarkus.platform.artifact-id}</artifactId>
+                <version>${quarkus.platform.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.acme</groupId>
+            <artifactId>quarkus-greeting</artifactId>
+        </dependency>
+    </dependencies>
+...
+```
+
+
+From a extension developper standpoint this publication strategy is very handy and fast but Quarkus wants to go one step further by also ensure a reliability of the ecosystem for the people who will use the extensions.
+Think about it, we all had a poor Developper eXperience with an unmaintained library, a incompatibility between dependencies (and we don't even talk about legal issues).
+That's why there is the Quarkus Platform.
+
+*Quarkus Platform*
+
+Quarkus proposes a `quarkus-platform-bom` which is a certified list of extensions placed under the Quarkus Platform label. From an application developper, the objectives of the platform are:
+
+* To guarantee a supportability of the extension (bugfix, security issues, dependency upgrades)
+* To ease the extension discovery through the Quarkus CLI or https://code.quarkus.io/
+* To ensure a global consitency of the extension ecosystem
+
+Note: As for now, the process to propose a new extension is not defined yet. Your best chance is to present your extension on the https://groups.google.com/forum/#!forum/quarkus-dev[Quarkus Google Group] or on https://quarkusio.zulipchat.com/[Zulip] and wait for an official invitation to join the Quarkus Platform.
+
+== Conclusion
+
+Creating new extension can look impressing at first but once you understood the Quarkus game-changer paradigm (build time vs. runtime) the structure makes sense.
```

Comment:
`can look impressing` should probably be re-phrased to something like "may appear to be an intricate task".